app/controllers/pieces_controller.rb

```diff
@@ -5,20 +5,30 @@ class PiecesController < ApplicationController
   def update
     x = params[:piece][:x_coordinate].to_i
     y = params[:piece][:y_coordinate].to_i
-    
-    if @piece.move_to! x, y
-      flash[:success] = "#{@piece.color.capitalize} #{@piece.type} move to X#{x}/Y#{y} was valid" if @piece.update_attributes(piece_params.merge(has_moved?: true))
+  
+    unless @game.is_finished?
```

Comment:
There are times when it's not a bad thing to do, but in this case it's a fine change.